lib/core/src/Ouroboros/Network/Client/Wallet.hs

```diff
@@ -131,8 +151,10 @@ import qualified Ouroboros.Network.Protocol.LocalStateQuery.Client as LSQ
 chainSyncFollowTip
     :: forall m block era. (Monad m)
     => (block -> era)
-    -> (Maybe era -> Tip block -> m ())
+    -> (era -> Tip block -> m ())
```

Comment:
I am afraid this may lead to nasty errors. The reason why I initially used a `Maybe` here was to stil allow `Tip` updates. This is perhaps confusing and we may consider having 2 distinct callbacks then: one for the `Tip` and one for the `era`.

Reply:
What kind of nasty errors? It simplifies things, and I didn't see how it could be bad.

That said, now with `STM` `guard` it might be pretty ok to keep the `Maybe`…

Reply:
If we rollback, it means that for a while, we'll be using a tip that is in the future.

Reply:
Should be done -> 30c569d